From b56334be661ae0c5d8691545046c0277688ec5db Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:18 -0400
Subject: [tools] perf/script: Print addr by default for BTS

Message-id: <1408462094-14194-210-git-send-email-jolsa@redhat.com>
Patchwork-id: 88215
O-Subject: [PATCH RHEL7.1 BZ1131394 209/325] perf script: Print addr by default for BTS
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 243be3dd7c14454899e51334a6d66d29a41ae6ab
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Fri Oct 18 15:29:14 2013 +0300

description
===========
The addr field is not displayed by default for hardware events, however
for branch events it is the target of the branch so for BTS display it
by default if it was recorded.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-script.c b/tools/perf/builtin-script.c
index 9c333ff..ebb2b5f 100644
--- a/tools/perf/builtin-script.c
+++ b/tools/perf/builtin-script.c
@@ -409,7 +409,9 @@ static void print_sample_bts(union perf_event *event,
  printf(" => ");
 
  /* print branch_to information */
- if (PRINT_FIELD(ADDR))
+ if (PRINT_FIELD(ADDR) ||
+     ((evsel->attr.sample_type & PERF_SAMPLE_ADDR) &&
+      !output[attr->type].user_set))
   print_sample_addr(event, sample, machine, thread, attr);
 
  printf("\n");
-- 
1.7.1